image-rs/src/image.rs

```diff
@@ -197,22 +240,48 @@ impl ImageClient {
                 None => bail!("Invalid authentication info ({:?})", auth_info),
             },
             None => match &self.registry_auth {
-                Some(registry_auth) => registry_auth.credential_for_reference(&reference).await?,
+                Some(registry_auth) => {
+                    registry_auth
+                        .credential_for_reference(&task.image_reference)
+                        .await?
+                }
                 None => {
                     info!("Use Anonymous image registry auth");
                     RegistryAuth::Anonymous
                 }
             },
         };
 
+        let mut client_config = ClientConfig::default();
+        if task.use_http {
+            client_config.protocol = ClientProtocol::Http;
+        }
+
+        if let Some(no_proxy) = &self.config.skip_proxy_ips {
+            client_config.no_proxy = Some(no_proxy.clone())
+        }
+
+        if let Some(https_proxy) = &self.config.image_pull_proxy {
```

Comment:
If I understand this correctly, this is kind of curious. A user could specify an https proxy in their `ImageConfig` even though, with the new registry config that you've added, they might want to pull from a local registry with http, for example. Maybe it's better to at least issue a warning of misconfiguration here? (Otherwise what happens? The `oci` crate will happily oblige and try to go through the https proxy before using the local registry you specified?)

Reply:
Yes, this is indeed a corner case, but it does fit the logic of the configuration. I'll add a warning here

Reply:
Added the warning https://github.com/confidential-containers/guest-components/pull/944/files#diff-d753fb19cbbddd2cb199db08ea89c5ebe30e51c0edf69dc4414c36390eae9c44R276-R281